app/questionnaire/path_finder.py

```diff
@@ -207,7 +172,10 @@ def get_next_location(self, current_location):
         :param current_location:
         :return: The next location as a dict
         """
-        location_path = self.get_location_path(current_location.group_id, current_location.group_instance)
+        location_path = self.get_routing_path(current_location.group_id, current_location.group_instance)
+        if self.can_reach_summary(location_path):
+            location_path.append(Location(SchemaHelper.get_last_group_id(self.survey_json), 0, 'thank-you'))
```

Comment:
It's the only place now using it so don't think it needs to be more global at this moment. I have refactored this out for clarity though.